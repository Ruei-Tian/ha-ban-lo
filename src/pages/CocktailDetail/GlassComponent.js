import React from 'react';
import {
  GlassOfBrandy,
  GlassOfChampagne,
  GlassOfHighball,
  GlassOfMartini,
  GlassOfShot,
  GlassOfUFO,
  GlassOfWhisky
} from '../../components/SVG';

const GlassComponent = (props) => {
  const { glassType, colors } = props;
  let glassSVG = null;
  if (glassType === 'GlassOfBrandy') {
    glassSVG = <GlassOfBrandy mainColor={colors} />;
    return glassSVG;
  }
  if (glassType === 'GlassOfChampagne') {
    glassSVG = <GlassOfChampagne mainColor={colors} />;
    return glassSVG;
  }
  if (glassType === 'GlassOfHighball') {
    glassSVG = <GlassOfHighball mainColor={colors} />;
    return glassSVG;
  }
  if (glassType === 'GlassOfMartini') {
    glassSVG = <GlassOfMartini mainColor={colors} />;
    return glassSVG;
  }
  if (glassType === 'GlassOfShot') {
    glassSVG = <GlassOfShot mainColor={colors} />;
    return glassSVG;
  }
  if (glassType === 'GlassOfUFO') {
    glassSVG = <GlassOfUFO mainColor={colors} />;
    return glassSVG;
  }
  if (glassType === 'GlassOfWhisky') {
    glassSVG = <GlassOfWhisky mainColor={colors} />;
    return glassSVG;
  }
};

export default GlassComponent;
