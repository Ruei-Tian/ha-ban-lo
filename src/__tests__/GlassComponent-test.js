import React from 'react';
import renderer from 'react-test-renderer';
import GlassComponent from '../pages/CocktailDetail/GlassComponent';

const colorArray = [[46, 53, 44], [192, 63, 7], [224, 215, 201]];

describe('matched glass svg base on glass type', () => {
  it('Champagne', () => {
    const tree = renderer
      .create(<GlassComponent glassType="GlassOfChampagne" colors={colorArray} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Brandy', () => {
    const tree = renderer
      .create(<GlassComponent glassType="GlassOfBrandy" colors={colorArray} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Highball', () => {
    const tree = renderer
      .create(<GlassComponent glassType="GlassOfHighball" colors={colorArray} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Martini', () => {
    const tree = renderer
      .create(<GlassComponent glassType="GlassOfMartini" colors={colorArray} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Shot', () => {
    const tree = renderer
      .create(<GlassComponent glassType="GlassOfShot" colors={colorArray} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('UFO', () => {
    const tree = renderer
      .create(<GlassComponent glassType="GlassOfUFO" colors={colorArray} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Whisky', () => {
    const tree = renderer
      .create(<GlassComponent glassType="GlassOfWhisky" colors={colorArray} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
})
