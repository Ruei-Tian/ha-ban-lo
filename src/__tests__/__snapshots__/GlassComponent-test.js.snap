// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matched glass svg base on glass type Brandy 1`] = `
<svg
  height="200px"
  id="圖層_1"
  version="1.1"
  viewBox="0 0 336.854 399.479"
  width="150px"
  x="0px"
  xlink="http://www.w3.org/1999/xlink"
  xmlns="http://www.w3.org/2000/svg"
  y="0px"
>
  <defs>
    <linearGradient
      id="color-gradient"
      spreadMethod="pad"
      x1="0"
      x2="0"
      y1="0"
      y2="0.8"
    >
      <stop
        offset="0.0"
        style={
          Object {
            "stopColor": "rgb(46, 53, 44)",
            "stopOpacity": 1,
          }
        }
      />
      <stop
        offset="0.3"
        style={
          Object {
            "stopColor": "rgb(192, 63, 7)",
            "stopOpacity": 1,
          }
        }
      />
      <stop
        offset="0.7"
        style={
          Object {
            "stopColor": "rgb(224, 215, 201)",
            "stopOpacity": 1,
          }
        }
      />
      <animate
        attributeName="y1"
        dur="2s"
        from="1"
        to="0"
      />
    </linearGradient>
  </defs>
  <rect
    fill="none"
    height="6.766"
    stroke="#070001"
    strokeMiterlimit="10"
    strokeWidth="5"
    width="183.549"
    x="76.652"
    y="380.103"
  />
  <rect
    fill="none"
    height="3.045"
    stroke="#070001"
    strokeMiterlimit="10"
    strokeWidth="5"
    width="183.549"
    x="76.652"
    y="393.935"
  />
  <line
    fill="none"
    stroke="#070001"
    strokeMiterlimit="10"
    strokeWidth="40"
    x1="168.427"
    x2="168.427"
    y1="281.966"
    y2="380.103"
  />
  <path
    d="M331.825,184.826
 C312.688,116.413,289.933,2.5,289.933,2.5H168.427h-0.001H46.921c0,0-22.756,113.913-41.893,182.326
 c-19.048,68.096,72.864,98.804,163.398,99.088l0,0l0,0c0.001,0,0.001,0,0.001,0l0,0C258.962,283.63,350.872,252.922,331.825,184.826
 z"
    fill="url(#color-gradient)"
    stroke="#070001"
    strokeMiterlimit="10"
    strokeWidth="5"
  />
</svg>
`;

exports[`matched glass svg base on glass type Champagne 1`] = `
<svg
  height="200px"
  id="圖層_1"
  version="1.1"
  viewBox="0 0 148.75 449.305"
  width="150px"
  x="0px"
  xlink="http://www.w3.org/1999/xlink"
  xmlns="http://www.w3.org/2000/svg"
  y="0px"
>
  <defs>
    <linearGradient
      id="color-gradient"
      spreadMethod="pad"
      x1="0"
      x2="0"
      y1="0"
      y2="0.8"
    >
      <stop
        offset="0.0"
        style={
          Object {
            "stopColor": "rgb(224, 215, 201)",
            "stopOpacity": 1,
          }
        }
      />
      <stop
        offset="0.3"
        style={
          Object {
            "stopColor": "rgb(192, 63, 7)",
            "stopOpacity": 1,
          }
        }
      />
      <stop
        offset="0.7"
        style={
          Object {
            "stopColor": "rgb(46, 53, 44)",
            "stopOpacity": 1,
          }
        }
      />
      <animate
        attributeName="y1"
        dur="2s"
        from="1"
        to="0"
      />
    </linearGradient>
  </defs>
  <g>
    <rect
      fill="none"
      height="6.766"
      stroke="#040000"
      strokeMiterlimit="10"
      strokeWidth="5"
      width="125.6"
      x="11.576"
      y="429.928"
    />
    <rect
      fill="none"
      height="3.045"
      stroke="#040000"
      strokeMiterlimit="10"
      strokeWidth="5"
      width="125.6"
      x="11.576"
      y="443.76"
    />
    <line
      fill="none"
      stroke="#040000"
      strokeMiterlimit="10"
      strokeWidth="5"
      x1="74.375"
      x2="74.375"
      y1="292.842"
      y2="429.928"
    />
    <path
      d="M2.5,2.5v218.466
 c0,39.696,32.18,71.876,71.875,71.876c39.696,0,71.875-32.18,71.875-71.876l0,0V2.5H2.5z"
      fill="url(#color-gradient)"
      stroke="#040000"
      strokeMiterlimit="10"
      strokeWidth="5"
    />
  </g>
</svg>
`;

exports[`matched glass svg base on glass type Highball 1`] = `
<svg
  height="200px"
  id="圖層_1"
  version="1.1"
  viewBox="0 0 209.5 477.5"
  width="200px"
  x="0px"
  xlink="http://www.w3.org/1999/xlink"
  xmlns="http://www.w3.org/2000/svg"
  y="0px"
>
  <defs>
    <linearGradient
      id="color-gradient"
      spreadMethod="pad"
      x1="0"
      x2="0"
      y1="0"
      y2="0.8"
    >
      <stop
        offset="0.0"
        style={
          Object {
            "stopColor": "rgb(224, 215, 201)",
            "stopOpacity": 1,
          }
        }
      />
      <stop
        offset="0.3"
        style={
          Object {
            "stopColor": "rgb(192, 63, 7)",
            "stopOpacity": 1,
          }
        }
      />
      <stop
        offset="0.7"
        style={
          Object {
            "stopColor": "rgb(46, 53, 44)",
            "stopOpacity": 1,
          }
        }
      />
      <animate
        attributeName="y1"
        dur="2s"
        from="1"
        to="0"
      />
    </linearGradient>
  </defs>
  <rect
    fill="url(#color-gradient)"
    height="426.5"
    stroke="#000000"
    strokeMiterlimit="10"
    strokeWidth="5"
    width="204.5"
    x="2.5"
    y="5"
  />
  <rect
    fill="none"
    height="16"
    stroke="#000000"
    strokeMiterlimit="10"
    strokeWidth="5"
    width="204.5"
    x="2.5"
    y="444.5"
  />
  <rect
    fill="none"
    height="5"
    stroke="#000000"
    strokeMiterlimit="10"
    strokeWidth="5"
    width="204.5"
    x="2.5"
    y="472.5"
  />
  <g>
    <rect
      fill="none"
      height="59.528"
      stroke="#FFFFFF"
      strokeMiterlimit="10"
      strokeWidth="3"
      transform="matrix(0.8767 -0.4811 0.4811 0.8767 -54.4549 44.7975)"
      width="59.527"
      x="30.376"
      y="98.836"
    />
    <rect
      fill="none"
      height="59.528"
      stroke="#FFFFFF"
      strokeMiterlimit="10"
      strokeWidth="3"
      transform="matrix(0.9747 0.2236 -0.2236 0.9747 40.6344 -30.3012)"
      width="59.527"
      x="124.345"
      y="134.502"
    />
    <rect
      fill="none"
      height="59.528"
      stroke="#FFFFFF"
      strokeMiterlimit="10"
      strokeWidth="3"
      transform="matrix(0.9837 -0.1796 0.1796 0.9837 -39.001 17.6033)"
      width="59.527"
      x="47.932"
      y="194.379"
    />
    <rect
      fill="none"
      height="59.528"
      stroke="#FFFFFF"
      strokeMiterlimit="10"
      strokeWidth="3"
      transform="matrix(0.3489 0.9372 -0.9372 0.3489 368.0164 50.0515)"
      width="59.526"
      x="118.224"
      y="260.118"
    />
    <rect
      fill="none"
      height="59.527"
      stroke="#FFFFFF"
      strokeMiterlimit="10"
      strokeWidth="3"
      transform="matrix(0.6182 0.786 -0.786 0.6182 285.5562 75.4453)"
      width="59.528"
      x="35.35"
      y="301.914"
    />
  </g>
</svg>
`;

exports[`matched glass svg base on glass type Martini 1`] = `
<svg
  height="200px"
  id="圖層_1"
  version="1.1"
  viewBox="0 0 263.634 399.388"
  width="150px"
  x="0px"
  xlink="http://www.w3.org/1999/xlink"
  xmlns="http://www.w3.org/2000/svg"
  y="0px"
>
  <defs>
    <linearGradient
      id="color-gradient"
      spreadMethod="pad"
      x1="0"
      x2="0"
      y1="0"
      y2="0.8"
    >
      <stop
        offset="0.0"
        style={
          Object {
            "stopColor": "rgb(46, 53, 44)",
            "stopOpacity": 1,
          }
        }
      />
      <stop
        offset="0.3"
        style={
          Object {
            "stopColor": "rgb(192, 63, 7)",
            "stopOpacity": 1,
          }
        }
      />
      <stop
        offset="0.7"
        style={
          Object {
            "stopColor": "rgb(224, 215, 201)",
            "stopOpacity": 1,
          }
        }
      />
      <animate
        attributeName="y1"
        dur="2s"
        from="1"
        to="0"
      />
    </linearGradient>
  </defs>
  <rect
    fill="none"
    height="6.766"
    stroke="#040000"
    strokeMiterlimit="10"
    strokeWidth="5"
    width="125.599"
    x="69.018"
    y="380.011"
  />
  <rect
    fill="none"
    height="3.044"
    stroke="#040000"
    strokeMiterlimit="10"
    strokeWidth="5"
    width="125.599"
    x="69.018"
    y="393.844"
  />
  <polygon
    fill="url(#color-gradient)"
    points="131.817,242.925 68.376,156.486
 4.936,70.048 131.817,70.048 258.698,70.048 195.258,156.486 "
    stroke="#040000"
    strokeMiterlimit="10"
    strokeWidth="5"
  />
  <line
    fill="none"
    stroke="#040000"
    strokeMiterlimit="10"
    strokeWidth="5"
    x1="131.817"
    x2="131.817"
    y1="242.925"
    y2="380.011"
  />
  <polyline
    fill="none"
    points="258.609,3.938 197.197,14.889
 145.668,197.938 "
    stroke="#000000"
    strokeMiterlimit="10"
    strokeWidth="8"
  />
</svg>
`;

exports[`matched glass svg base on glass type Shot 1`] = `
<svg
  height="150px"
  id="圖層_1"
  version="1.1"
  viewBox="9.584 -37.864 190.332 300.864"
  width="100px"
  x="0px"
  xlink="http://www.w3.org/1999/xlink"
  xmlns="http://www.w3.org/2000/svg"
  y="0px"
>
  <defs>
    <linearGradient
      id="color-gradient"
      spreadMethod="pad"
      x1="0"
      x2="0"
      y1="0"
      y2="0.8"
    >
      <stop
        offset="0.0"
        style={
          Object {
            "stopColor": "rgb(224, 215, 201)",
            "stopOpacity": 1,
          }
        }
      />
      <stop
        offset="0.3"
        style={
          Object {
            "stopColor": "rgb(192, 63, 7)",
            "stopOpacity": 1,
          }
        }
      />
      <stop
        offset="0.7"
        style={
          Object {
            "stopColor": "rgb(46, 53, 44)",
            "stopOpacity": 1,
          }
        }
      />
      <animate
        attributeName="y1"
        dur="2s"
        from="1"
        to="0"
      />
    </linearGradient>
  </defs>
  <polygon
    fill="url(#color-gradient)"
    points="12.5,-35.364 197,-35.364 157,223.394
52.5,223.394 "
    stroke="#000000"
    strokeMiterlimit="10"
    strokeWidth="5"
  />
  <rect
    fill="none"
    height="15.731"
    stroke="#000000"
    strokeMiterlimit="10"
    strokeWidth="5"
    width="109.434"
    x="50.033"
    y="232.75"
  />
  <rect
    fill="none"
    height="3.599"
    stroke="#000000"
    strokeMiterlimit="10"
    strokeWidth="5"
    width="109.434"
    x="50.033"
    y="256.901"
  />
</svg>
`;

exports[`matched glass svg base on glass type UFO 1`] = `
<svg
  height="200px"
  id="圖層_1"
  version="1.1"
  viewBox="0 0 316.634 279.318"
  width="150px"
  x="0px"
  xlink="http://www.w3.org/1999/xlink"
  xmlns="http://www.w3.org/2000/svg"
  y="0px"
>
  <defs>
    <linearGradient
      id="color-gradient"
      spreadMethod="pad"
      x1="0"
      x2="0"
      y1="0"
      y2="0.8"
    >
      <stop
        offset="0.0"
        style={
          Object {
            "stopColor": "rgb(46, 53, 44)",
            "stopOpacity": 1,
          }
        }
      />
      <stop
        offset="0.3"
        style={
          Object {
            "stopColor": "rgb(192, 63, 7)",
            "stopOpacity": 1,
          }
        }
      />
      <stop
        offset="0.7"
        style={
          Object {
            "stopColor": "rgb(224, 215, 201)",
            "stopOpacity": 1,
          }
        }
      />
      <animate
        attributeName="y1"
        dur="2s"
        from="1"
        to="0"
      />
    </linearGradient>
  </defs>
  <rect
    fill="none"
    height="6.766"
    stroke="#040000"
    strokeMiterlimit="10"
    strokeWidth="5"
    width="125.6"
    x="95.518"
    y="259.941"
  />
  <rect
    fill="none"
    height="3.045"
    stroke="#040000"
    strokeMiterlimit="10"
    strokeWidth="5"
    width="125.6"
    x="95.518"
    y="273.773"
  />
  <line
    fill="none"
    stroke="#040000"
    strokeMiterlimit="10"
    strokeWidth="5"
    x1="158.317"
    x2="158.317"
    y1="122.855"
    y2="259.941"
  />
  <path
    d="M2.5,2.5v20.951
 c0,15.173,29.212,28.485,73.141,35.994h-1.544v20.951c0,23.449,37.706,42.459,84.22,42.459c46.515,0,84.222-19.01,84.222-42.459
 V59.445h-1.545c43.929-7.509,73.141-20.821,73.141-35.994V2.5H2.5z"
    fill="url(#color-gradient)"
    stroke="#040000"
    strokeMiterlimit="10"
    strokeWidth="5"
  />
</svg>
`;

exports[`matched glass svg base on glass type Whisky 1`] = `
<svg
  height="150px"
  id="圖層_1"
  version="1.1"
  viewBox="0 0 265 263"
  width="150px"
  x="0px"
  xlink="http://www.w3.org/1999/xlink"
  xmlns="http://www.w3.org/2000/svg"
  y="0px"
>
  <defs>
    <linearGradient
      id="color-gradient"
      spreadMethod="pad"
      x1="0"
      x2="0"
      y1="0"
      y2="0.8"
    >
      <stop
        offset="0.0"
        style={
          Object {
            "stopColor": "rgb(224, 215, 201)",
            "stopOpacity": 1,
          }
        }
      />
      <stop
        offset="0.3"
        style={
          Object {
            "stopColor": "rgb(192, 63, 7)",
            "stopOpacity": 1,
          }
        }
      />
      <stop
        offset="0.7"
        style={
          Object {
            "stopColor": "rgb(46, 53, 44)",
            "stopOpacity": 1,
          }
        }
      />
      <animate
        attributeName="y1"
        dur="2s"
        from="1"
        to="0"
      />
    </linearGradient>
  </defs>
  <rect
    fill="url(#color-gradient)"
    height="212"
    stroke="#000000"
    strokeMiterlimit="10"
    strokeWidth="5"
    width="260"
    x="2.5"
    y="2.5"
  />
  <rect
    fill="none"
    height="16"
    stroke="#000000"
    strokeMiterlimit="10"
    strokeWidth="5"
    width="260"
    x="2.5"
    y="227.5"
  />
  <rect
    fill="none"
    height="5"
    stroke="#000000"
    strokeMiterlimit="10"
    strokeWidth="5"
    width="260"
    x="2.5"
    y="255.5"
  />
  <rect
    fill="none"
    height="53.864"
    stroke="#FFFFFF"
    strokeMiterlimit="10"
    strokeWidth="2"
    transform="matrix(0.9998 0.0223 -0.0223 0.9998 4.0071 -2.2595)"
    width="57.932"
    x="74.568"
    y="152"
  />
  <rect
    fill="none"
    height="53.864"
    stroke="#FFFFFF"
    strokeMiterlimit="10"
    strokeWidth="2"
    transform="matrix(0.9433 0.332 -0.332 0.9433 39.1181 -11.9514)"
    width="57.932"
    x="25.568"
    y="81.568"
  />
  <rect
    fill="none"
    height="53.863"
    stroke="#FFFFFF"
    strokeMiterlimit="10"
    strokeWidth="2"
    transform="matrix(0.7981 -0.6025 0.6025 0.7981 -57.2012 140.1522)"
    width="57.932"
    x="151.568"
    y="128.5"
  />
</svg>
`;
